docs(getting-started): revamp Getting Started

hassankhan opened the pull request.

Rory-Cube reviewed (approved):
Looks good!

igorlukanin commented on docs/content/Getting-Started/Overview.mdx:
Since this page should convince readers to choose the best experience for them, we should be very clear about the benefits of these two options. I think it would be great to add more details on "why Cube Cloud" here

igorlukanin commented on docs/content/Getting-Started/Cloud/01-Overview.mdx:
```diff
@@ -0,0 +1,22 @@
+isDisableFeedbackBlock: true
+---
+
+Learn how to set up a new deployment on Cube Cloud using a sample database.
```
This is a great example of a place where a short paragraph that explains "why should one do that" would be really cool to have.

igorlukanin commented on docs/content/Getting-Started/Cloud/02-Create-a-deployment.mdx:
```diff
@@ -0,0 +1,104 @@
+
+## Create deployment
+
+The first step in creating a deployment is to sign in to your Cube Cloud
+account. After signing in, click the **Create Deployment** button:
```
How exactly would one sign up? I think it would be great to add a link to the sign-up page

igorlukanin commented on docs/content/Getting-Started/Cloud/02-Create-a-deployment.mdx:
```diff
@@ -0,0 +1,104 @@
+
+</InfoBox>
+
+To do this, first select the data source:
```
In the info box above, it's not specified that the sample database is Postgres. So, it's unclear what should one select here. Only later the image says "Postgres."

igorlukanin commented on docs/content/Getting-Started/Cloud/02-Create-a-deployment.mdx:
```diff
@@ -0,0 +1,104 @@
+
+If you run into issues here, make sure to allow inbound connections from Cube
+Cloud IPs to access your database. This means you need to enable these IPs in
+your firewall. If you are using AWS, this would mean adding a security group
+with allowed IPs.
```
It would be great to add a link to AWS docs on adding a security group with allowed IPs.

igorlukanin commented on docs/content/Getting-Started/Cloud/03-Generate-models.mdx:
```diff
@@ -0,0 +1,96 @@
+menuOrder: 4
+isDisableFeedbackBlock: true
+---
+
```
Again, a really nice place to have a paragraph explaining why should one do that.

igorlukanin commented on docs/content/Getting-Started/Cloud/03-Generate-models.mdx:
```diff
@@ -0,0 +1,96 @@
+
+## Measures and dimensions
+
+Next, check the detected measures and dimensions to ensure correctness. After
```
Sounds a bit strange: if the sample dataset is used, it should work out of the box; if a reader uses her own dataset, what should she look at, exactly? What can be incorrect?

igorlukanin commented on docs/content/Getting-Started/Cloud/03-Generate-models.mdx:
```diff
@@ -0,0 +1,96 @@
+
+## Review
+
+Finally, the last screen allows you to review all the detected cubes and their
+corresponding measures, dimensions and joins. If everything looks good, click
+**Confirm & Generate**:
```
There's also an option to select data model language there: YAML or JavaScript. I think we need to explain the trade-off and provide a link to relevant sections.

igorlukanin commented on docs/content/Getting-Started/Cloud/04-Query-Cube-API.mdx:
```diff
@@ -0,0 +1,121 @@
+
+## Developer Playground
+
+The Playground is a web-based tool that allows you to query your Cube data
```
I think "Cube data" is not a very precise term. Maybe remove "Cube"?

igorlukanin commented on docs/content/Getting-Started/Cloud/04-Query-Cube-API.mdx:
```diff
@@ -0,0 +1,121 @@
+run the query. Let's do this for the `Orders` cube you generated in the previous
+step.
+
+Click **+ Measure** to display the available measures and add `Orders.count`,
+then click **+ Dimension** for available dimensions and add `Orders.status`.
```
I would recommend to do this in two steps: first, select only a dimension, show results; then, add a measure, show the changes. It should be more telling than composing a "complex" query all it once.

igorlukanin commented on docs/content/Getting-Started/Cloud/04-Query-Cube-API.mdx:
```diff
@@ -0,0 +1,121 @@
+</div>
+
+Since the Cube SQL API is Postgres-compatible, please make sure to select
+**Postgres** as the database type when connecting from BI tools.
```
No link to the next section here.

igorlukanin commented on docs/content/Getting-Started/Cloud/04-Query-Cube-API.mdx:
```diff
@@ -0,0 +1,121 @@
+
+<div style="text-align: center">
+  <img
+    src="https://ucarecdn.com/ffbaeb08-dfa5-4b64-b578-41206941f41b/"
```
Looks like a very outdated screenshot, we now have much more data consumers in there.

igorlukanin commented on docs/content/Getting-Started/Cloud/04-Query-Cube-API.mdx:
```diff
@@ -0,0 +1,121 @@
+
+On the Overview page, click **How to connect**, which will open a modal with
+instructions on different ways to connect to the SQL API. Navigate to the SQL
+API tab and enable the SQL API.
```
Similarly to sections on REST API and GraphQL API, it would be great to demonstrate a sample SQL query.

igorlukanin commented on docs/content/Getting-Started/Cloud/05-Add-a-pre-aggregation.mdx:
```diff
@@ -0,0 +1,47 @@
+menuOrder: 6
+isDisableFeedbackBlock: true
+---
+
```
Again, let's tell why. Totally unclear why would one bother to do that.

igorlukanin commented on docs/content/Getting-Started/Cloud/05-Add-a-pre-aggregation.mdx:
```diff
@@ -0,0 +1,47 @@
+  />
+</div>
+
+Just above the results, click on **Query was not accelerated with
```
Let's put an emphasis on the query latency. It's three-digit in ms, right?

hassankhan replied:
Can we discuss this in a Slack huddle?

igorlukanin commented on docs/content/Getting-Started/Cloud/05-Add-a-pre-aggregation.mdx:
```diff
@@ -0,0 +1,47 @@
+
+The Rollup Designer will automatically suggest a pre-aggregation for the query;
+simply click **Add to the Data Schema** and then retry the query in the
+Playground. This time, the query should be accelerated with a pre-aggregation:
```
Let's put an emphasis on the query latency. It's now two-digit in ms, much better.

hassankhan replied:
Can we discuss this in a Slack huddle?

igorlukanin commented on docs/content/Getting-Started/Cloud/06-Learn-more.mdx:
```diff
@@ -0,0 +1,26 @@
+subCategory: Cube Cloud
+menuOrder: 7
+isDisableFeedbackBlock: true
+---
```
I think much more might be added here. Team collaboration in Cube Cloud, query debugging, etc. Basically everything from the Cloud section of docs (I also think it shoudld be merged, we don't really need separate "Cube Cloud docs".)

igorlukanin commented on docs/content/Getting-Started/Core/02-Create-a-project.mdx:
````diff
@@ -0,0 +1,102 @@
+To create a new project, run the following command from your terminal:
+
+```bash
+npx cubejs-cli create my-first-cube-project
````
I'm absolutely positive we should have no "npx" in the "getting started" sequence. My suggestion is to restore the `docker-compose.yml` file from the deleted docs page and recommend getting started with Docker Compose. It's a fully transparent and extendable way. Also, it lets us, later in docs, suggest to extend that same `docker-compose.yml` file with additional services, e.g., Cube Store.

igorlukanin commented on docs/content/Getting-Started/Core/02-Create-a-project.mdx:
```diff
@@ -0,0 +1,102 @@
+
+<br />
+
+| Field    | Value              |
```
I think credentials might be move a few paragraphs down where they are actually needed.

igorlukanin replied:
Also, let's specify it's Postgres.

igorlukanin commented on docs/content/Getting-Started/Core/03-Create-models.mdx:
```diff
@@ -1,25 +1,34 @@
-menuOrder: 1
-proofread: 06/18/2019
+title: Create models
+permalink: /getting-started/core/create-models
```
This page has no counterpart in the "Cube Cloud" sequence and I honestly doubt it should be a part of "Cube Core" sequence as well. Looks like something that belongs to "data modeling" section of the docs, not here. I suggest (re)moving this page.

igorlukanin commented on docs/content/Getting-Started/Core/04-Query-Cube-API.mdx:
```diff
@@ -0,0 +1,98 @@
+
+## SQL API
+
+To enable the SQL API, we first need to add a few new environment variables to
```
These variables could/should be added in the default `docker-compose.yml` file, so there's no need to have this part here.

hassankhan replied:
The default setup with Docker Compose automatically creates a `.env` file with all the database credentials, that's why we're adding it there

igorlukanin commented on docs/content/Getting-Started/Core/04-Query-Cube-API.mdx:
````diff
@@ -0,0 +1,98 @@
+CUBEJS_SQL_PASSWORD=mypassword
+```
+
+If you're running Cube through Docker Compose, remember to add a port mapping to
+the Cube service for `CUBEJS_PG_SQL_PORT`:
+
+```yaml
+services:
+  cube:
+    image: cubejs/cube
+    ports:
+      - 5432:5432 # Cube SQL API
+```
+
+Or if running Docker from command line:
+
+```bash
+docker run -p 4000:4000 -p 5432:5432 \
+  -v ${PWD}:/cube/conf \
+  -e CUBEJS_DEV_MODE=true \
+  -e CUBEJS_PG_SQL_PORT=5432 \
+  cubejs/cube
+```
````
There should be no "if", let's just stick with Docker Compose.

igorlukanin commented on docs/content/Getting-Started/Core/04-Query-Cube-API.mdx:
```diff
@@ -0,0 +1,98 @@
+---
+title: Query Cube API
```
Similar page in the "Cube Cloud" sequence is called "Query Cube". I would make their names same, e.g., "Query Data."

igorlukanin commented:
@hassankhan Hassan, this is absolutely awesome work! Thanks for inviting me to do a review ❤️ 

I think that a refresh of the "getting started" section was long overdue. So happy that it's happening now.

Below are some comments that apply to the whole PR. Also, I'll leave comments for particular files/lines.

I think it would be great to...
1. prepend every page with a paragraph explaining why it is necessary to take that step so readers build not only an understanding what to do but also why to do
1. append a paragraph or list of links to other pages in docs to every page so readers have means to expand their understanding of Cube where necessary, e.g., learn mode about data modeling, APIs, data sources, etc. (ultimately, the goal should be to have every other section of the docs cross-linked from the "getting started" sequence)
1. review every (of two) getting started sequences so that readers are not unnecessary exposed to unknown terms, e.g., in the sections on data modeling, they don't face "cubes" without an explanation of what they actually are.

Also, few minor and very subjective things:
- I think screenshots with borders (or browser window borders) would be more easy to study.
- I think it would be great if images span the whole page (not they have huge side margins).
- We have a special "button" tag/style in blog (see [example](https://cube.dev/blog/observable-tutorial-javascript-native-data-notebook#embedding-observable-notebooks)); it would be great to have that in docs as well; we can ask @tenebricosa to help, I think.

hassankhan commented:
@igorlukanin I think I've addressed most of your feedback. With regards to some of the minor feedback:

> I think screenshots with borders (or browser window borders) would be more easy to study.

I have an example PR where we can add browser chrome via CSS, I'm not convinced there's value to us taking screenshots with all our tabs open 😅

> I think it would be great if images span the whole page (not they have huge side margins).

Agreed, we can probably fix that for all images in the docs in a separate PR

> We have a special "button" tag/style in blog (see [example](https://cube.dev/blog/observable-tutorial-javascript-native-data-notebook#embedding-observable-notebooks))

That would be _awesome_, but should probably be a part of a separate PR too

igorlukanin commented on docs/content/Getting-Started/Overview.mdx:
This icon seems weird. What can we have here instead of Git? Docker?

igorlukanin commented on docs/content/Schema/Getting-Started.mdx:
Why is this file removed? It seems unrelated to this PR.

igorlukanin replied:
Yep, makes total sense!

igorlukanin commented on docs/content/SQL-API/Getting-Started.mdx:
Why is this file removed? It seems unrelated to this PR.

igorlukanin replied:
Thanks for explaining, Hassan! Now I get it 😊

igorlukanin commented on docs/content/Getting-Started/Core/02-Create-a-project.mdx:
````diff
@@ -0,0 +1,125 @@
+touch docker-compose.yml
+```
+
+Open the `docker-compose.yml` file and add the following content:
````
What do you think about using this [`cat EOF`](https://linuxhint.com/what-is-cat-eof-bash-script/) technique and compressing it to a single block of commands?

hassankhan replied:
I tried it:

<img width="885" alt="image" src="https://user-images.githubusercontent.com/1781985/229145877-3e4cb766-01b0-406c-b2de-2020c8bac3a4.png">

IMO the syntax highlighting looks better when its a separate snippet

igorlukanin commented on docs/content/Getting-Started/Core/04-Query-data.mdx:
````diff
@@ -0,0 +1,102 @@
+To enable the SQL API, we first need to add a few new environment variables to
+the project's `.env` file:
+
+```bash
+CUBEJS_PG_SQL_PORT=5432
+CUBEJS_SQL_USER=myusername
+CUBEJS_SQL_PASSWORD=mypassword
+```
````
Why don't have them added to the `docker-compose.yml` file in the first place? It will make the flow more smooth.

hassankhan replied:
Referring to my comment on a resolved feedback: https://github.com/cube-js/cube/pull/6013#discussion_r1151179855

igorlukanin replied:
OK, that makes sense. Then, why don't we go with this `docker-compose.yml` file? We're putting `CUBEJS_DEV_MODE` there anyway.

```yml
version: '2.2'

services:
  cube:
    image: cubejs/cube:latest
    ports:
      - 4000:4000  # Playground
      - 5432:5432  # SQL API
    environment:
      - CUBEJS_DEV_MODE=true
      - CUBEJS_PG_SQL_PORT=5432
      - CUBEJS_SQL_USER=myusername
      - CUBEJS_SQL_PASSWORD=mypassword
    volumes:
      - .:/cube/conf
```

hassankhan replied:
IIRC We need the initial `CUBEJS_DEV_MODE=true` in the Docker Compose file so Cube launches the Database Connection Wizard; after the DB connection details are set up, they're saved in the `.env` file along with all other Cube-related variables.
